frontend/src/components/RequireValidation/index.js

```diff
@@ -0,0 +1,29 @@
+import { PropTypes } from "prop-types";
+import { useNavigate } from "react-router-dom";
+import { useEffect, useState } from "react";
+
+const RequireValidation = ({ children }) => {
+  const navigator = useNavigate();
+  const [render, setRender] = useState(null);
+
+  useEffect(() => {
+    if (
+      (children.type.componentName === "WheelPage" &&
+        localStorage.getItem("userData")) ||
+      (children.type.componentName === "PrizePage" &&
+        localStorage.getItem("prizeWon"))
+    ) {
+      setRender(render);
+    } else {
+      navigator("/");
+    }
+  }, [children.type.componentName, render, navigator]);
+
+  return render;
```

Comment:
Why can't we just return `children` here?

Reply:
I tried and it would skip the useEffect and try to render the page, so I had to find a work around

Reply:
I would instead set a loading flag, and either return children or null

Reply:
On it!